### PR TITLE
Update to version 2.2.24

### PR DESCRIPTION
**Pull request by danclewley**
Pinned RSGISLib to version `rsgislib 3.3.*` as it is quite closely tied with this release and won't work with older version (or possibly newer ones).

Re-rendered feedstock with `conda-smithy`.

@petebunting - do you want to have a quick check everything looks OK with this.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ requirements:
     - python
     - py6s
     - gdal 2.2.*
```
Comment (gillins) [inline recipe/meta.yaml]: I think you need to pin to `gdal 2.1.*` to match `rsgislib` etc. I think this is why the builds are failing....

**Review by gillins — APPROVED** [review]

**Comment by danclewley** [comment]
@gillins - any idea why the macOS builds haven't uploaded.

Looking at the [Travis log](https://travis-ci.org/conda-forge/arcsi-feedstock/jobs/287703666) the build finished with:

```
$ upload_or_check_non_existence ./recipe conda-forge --channel=main
Using Anaconda API: https://api.anaconda.org
No BINSTAR_TOKEN present, so no upload is taking place. The distribution just built is not already available for conda-forge.
The command "upload_or_check_non_existence ./recipe conda-forge --channel=main" exited with 0.
Done. Your build exited with 0.
```

Linux and Windows uploaded OK.

**Comment by gillins** [comment]
No - sorry I don't know. @ocefpaf might have some ideas. When I've seen this before I had to update `conda-smithy` but I see you are already using the latest (2.4.2)....

**Comment by gillins** [comment]
Those files are up there now - thanks @ocefpaf or whoever restarted/fixed the build :smile:

**Comment by ocefpaf** [comment]
I don't remember restarting this one in particular but I did restart tons of failed CI jobs recently :smile: 
Glad it is resolved though.

PS: to always rerender with latest conda-smithy without actually installing it see https://conda-forge.org/docs/webservice.html

**Comment by gillins** [comment]
That's very cool - I didn't know about that. Thanks.

**Pull request merged**